packages/atomic/src/utils/utils.ts

```diff
@@ -98,6 +98,34 @@ export function elementHasAncestorTag(
   return elementHasAncestorTag(parentElement, tagName);
 }
 
+export function closest<K extends keyof HTMLElementTagNameMap>(
+  element: Element | null,
+  selector: K
+): HTMLElementTagNameMap[K] | null;
+export function closest<K extends keyof SVGElementTagNameMap>(
+  element: Element | null,
+  selector: K
+): SVGElementTagNameMap[K] | null;
+export function closest<E extends Element = Element>(
```

Comment:
All of these are copied from the signature of the original `closest`. The first one is actually useful for `updateBreakpoints`.